auth/src/main/java/org/aerogear/auth/RoleKey.java

```diff
@@ -0,0 +1,55 @@
+package org.aerogear.auth;
+import java.util.Objects;
+import org.aerogear.auth.impl.UserPrincipalImpl;
+
+/**
+ * Stores a user's keycloak roles information
+ */
+
+public class RoleKey {
+
+    private String roleName;
+    private RoleType roleType;
+    private String clientID;
+
+    /**
+     * Creates a new RoleKey object
+     *
+     * @param role
+     * @param clientID
+     */
+    public RoleKey(final IRole role, final String clientID) {
+        roleName = role.getRoleName();
+        roleType = role.getRoleType();
+        this.clientID = clientID;
+    }
+
+    public RoleKey(final String roleName, final String clientId, final RoleType roleType){
```

Comment:
Missing a comment from one of my changes

Reply:
Nice catch, thanks.